packages/eslint-plugin/src/util/isArrayMethodCallWithPredicate.ts

```diff
@@ -0,0 +1,35 @@
+import { getConstrainedTypeAtLocation } from '@typescript-eslint/type-utils';
+import type {
+  ParserServicesWithTypeInformation,
+  TSESTree,
+} from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+
+const ARRAY_PREDICATE_FUNCTIONS = new Set([
+  'filter',
+  'find',
+  'findIndex',
+  'findLast',
+  'findLastIndex',
+  'some',
+  'every',
+]);
+
+export function isArrayMethodCallWithPredicate(
+  services: ParserServicesWithTypeInformation,
+  node: TSESTree.CallExpression,
+): boolean {
+  if (
+    node.callee.type === AST_NODE_TYPES.MemberExpression &&
+    node.callee.property.type === AST_NODE_TYPES.Identifier &&
+    ARRAY_PREDICATE_FUNCTIONS.has(node.callee.property.name)
```

Comment:
```suggestion
    ARRAY_PREDICATE_FUNCTIONS.has(getStaticMemberAccessValue(node.callee))
```

[Refactor] Now that https://github.com/typescript-eslint/typescript-eslint/pull/9836 is already merged, we can use a nice utility function for this!